### ISSUE TITLE
SSG Feature request - disable all generators but allow output.

### ISSUE THREAD

**Issue by ImATrackMan** [rerrahkr/BambooTracker#219]
The SSG allows you to disable sound output but still set volume. This results in a DC pop and is used famously by Konami for SCC soundtracks and in the PC-88 version of Snatcher. BT currently does not allow for output to be completely disabled and only allows for tone output at minimum.

Examples:
https://youtu.be/QpdifQ6tWwc?t=76
https://www.youtube.com/watch?v=zvl4Z34IST8

**Comment by rerrahkr** [comment]
About "Konami drums" (I have heard like this), it uses SSG-EG techniques, not only a DC pop.
Using the bass drum sound as an example, you can see that SSE-EG and tone are used at the same time.

![konami_drums](https://user-images.githubusercontent.com/26477882/83963124-81ac9a00-a8de-11ea-995f-9aad5fbffee6.png)

It is a bit complicated on BambooTracker, but it can be reproduced by setting the hardware envelope and tone with the Envelope editor of SSG instrument.

I also find [the page](https://cpu.8bitsize.com/3mhz/index/03/s-00.php) (Japanese) explaining about Konami drums.

**Comment by ImATrackMan** [comment]
There are three main versions of the driver. F1 Spirit driver, Gradius II: Gofer no Yabou driver, and Space Manbow driver (for Space Mambow, MG2: Solid Snake, and SD Snatcher). The first two versions use only a pop, especially F1 Spirit. I took a screenshot from Salamander but my new PC audio drivers add too much bass so after the pop there is a lot of added low echo.


![image](https://user-images.githubusercontent.com/47734563/83965178-c228fb00-a87f-11ea-9a67-aa0db7d4d0c8.png)

Famitracker copy produces the same output
![image](https://user-images.githubusercontent.com/47734563/83965210-f2709980-a87f-11ea-83ad-c5096f5d1efb.png)
![image](https://user-images.githubusercontent.com/47734563/83965095-1f707c80-a87f-11ea-944d-0b301debe88c.png)
![image](https://user-images.githubusercontent.com/47734563/83965123-534ba200-a87f-11ea-9504-c7724e01e7f4.png)

You can see that in Famitracker the output is disabled for one frame but volume is still set

**Comment by ImATrackMan** [comment]
I may be able to find old recordings of the drums from my old PC but I will have to look for them

**Comment by rerrahkr** [comment]
Thanks for much information. The example of FamiTracker was easy to understand. I will try to change Tone/Noise editor to allow both tone and noise to be turned off.

**Comment by rerrahkr** [comment]
> Famitracker copy produces the same output

[test.zip](https://github.com/rerrahkr/BambooTracker/files/4742219/test.zip)
I prepare 2 vgm made by BambooTracker, tone_off.vgm only turns off output in the first tick, tone_volume_off.vgm turns off output and volume after setting envelope.
I tested them on G.I.M.I.C and felt the second one was closer to your sample.

tone_off.vgm is upper, tone_volume_off.vgm is lower
![pic](https://user-images.githubusercontent.com/26477882/83971592-1b903900-a917-11ea-87dc-1a37d0bc8510.png)

Could you test and compare these waveforms in your environment?

**Comment by rerrahkr** [comment]
I added this feature at 31ebe3d371bab5fe3c890eb212347583e2902315.

**Comment by ImATrackMan** [comment]
I have only tested in BT and not on hardware so far, but I was confused by the implementation. I can confirm it works the same way in other trackers like 0cc-FT, but I am very confused by how I've been replicating the trick.

Thank you for implementing.

**Comment by rerrahkr** [comment]
I'm not sure what you mean (I'm sorry I'm not good at English), but the method I actually implemented is the one in the image above.

>  only turns off output in the first tick

DC pop does not occur just by turning off the tone and noise and setting the volume at the same time.
A technique I didn't implement, once BambooTracker set the volume once and then change it to 0, it seems to be able to reproduce this pop (image below), but I'm not sure what Konami's example or 0CC-FT actually does.

I implemented the current method that does not change the volume to 0, guessing from the behavior of 0CC-FT editor. But would the other method be better?
